packages/attestation-service/src/db.ts

```diff
@@ -22,11 +23,31 @@ export let kit: ContractKit
 export async function initializeKit() {
   if (kit === undefined) {
     kit = newKit(fetchEnv('CELO_PROVIDER'))
-    const blockNumber = await kit.web3.eth.getBlockNumber()
-    if (blockNumber === 0) {
-      throw new Error(
-        'Could not fetch latest block from web3 provider ' + fetchEnv('CELO_PROVIDER')
+    // Copied from @celo/cli/src/utils/helpers
+    try {
+      const syncProgress = await kit.web3.eth.isSyncing()
+      if (typeof syncProgress === 'boolean' && !syncProgress) {
+        const latestBlock: Block = await kit.web3.eth.getBlock('latest')
+        if (latestBlock && latestBlock.number > 0) {
+          // To catch the case in which syncing has happened in the past,
+          // has stopped, and hasn't started again, check for an old timestamp
+          // on the latest block
+          const ageOfBlock = Date.now() / 1000 - latestBlock.timestamp
+          if (ageOfBlock > 120) {
+            throw new Error(
+              `Latest block is ${ageOfBlock} seconds old, and syncing is not currently in progress`
+            )
+          }
+        }
+      } else {
+        throw new Error('Node is not syncing')
```

Comment:
Is this the right message? If `eth_syncing` returned a non-`false` then the node *is syncing*. Should this message read `'Node is not synced'` instead?

Reply:
Good catch. I've opened https://github.com/celo-org/celo-monorepo/pull/2372